backend/tests/unit/graphql/test_mutation_create.py

```diff
@@ -1093,3 +1093,51 @@ async def test_incorrect_peer_type_prevented(db: InfrahubDatabase, default_branc
         result.errors[0].message
         == f"""TestDog - {dog2.id} cannot be added to relationship, must be of type: ['TestPerson'] at owner"""
     )
+
+
+async def test_create_valid_datetime_success(db: InfrahubDatabase, default_branch, criticality_schema):
+    query = """
+    mutation {
+        TestCriticalityCreate(data: {name: { value: "HIGH"}, level: {value: 1}, time: {value: "2021-01-01T00:00:00Z"}}) {
+            ok
+            object {
+                id
+            }
+        }
+    }
+    """
+    gql_params = prepare_graphql_params(db=db, include_subscription=False, branch=default_branch)
+    result = await graphql(
+        schema=gql_params.schema,
+        source=query,
+        context_value=gql_params.context,
+        root_value=None,
+        variable_values={},
+    )
+    assert result.errors is None
+    assert result.data["TestCriticalityCreate"]["ok"] is True
```

Comment:
I'm going to merge this but just as a FYI (and for others) here result.data is an `Optional[dict]` and as such it could be that it's empty. It's mostly a matter of type-checking within the tests which is mostly disabled now and missing in multiple places but before asserting something under "result.data". we should have a line above it for `assert result.data` to ensure that this value is not `None`